openmc/mgxs/library.py

```diff
@@ -819,20 +820,22 @@ def get_xsdata(self, domain, xsdata_name, nuclide='total', xs_type='macro',
             name += '_' + nuclide
         name += '.' + xs_id
         xsdata = openmc.XSdata(name, self.energy_groups)
-        if order is 0:
-            xsdata.order = order
+
+        if order is None:
+            # Set the order to the Library's order (the defualt behavior)
+            xsdata.order = self.legendre_order
         else:
-            msg = 'Generating anisotropic scattering from openmc.Library' \
-                  'objects has not yet been implemented.'
-            raise NotImplementedError(msg)
+            # Set the order of the xsdata object to the minimum of
+            # the provided order or the Library's order.
+            xsdata.order = min(order, self.legendre_order)
```

Comment:
Print a warning message here if the two don't match. Also this behaviour should be explicitly pointed out in the docstring.